prowler/providers/aws/services/iam/iam_role_cross_service_confused_deputy_prevention/iam_role_cross_service_confused_deputy_prevention.py

```diff
@@ -50,6 +50,50 @@ def execute(self) -> Check_Report_AWS:
                                     ]
                                 )
                             )
+                            or (
+                                "StringEquals" in statement["Condition"]
+                                and "aws:PrincipalAccount"
+                                in statement["Condition"]["StringEquals"]
+                                and iam_client.audited_account
+                                in str(
+                                    statement["Condition"]["StringEquals"][
+                                        "aws:PrincipalAccount"
+                                    ]
+                                )
+                            )
+                            or (
+                                "StringLike" in statement["Condition"]
+                                and "aws:PrincipalAccount"
+                                in statement["Condition"]["StringLike"]
+                                and iam_client.audited_account
+                                in str(
+                                    statement["Condition"]["StringLike"][
+                                        "aws:PrincipalAccount"
+                                    ]
+                                )
+                            )
+                            or (
+                                "StringEquals" in statement["Condition"]
+                                and "aws:ResourceAccount"
+                                in statement["Condition"]["StringEquals"]
+                                and iam_client.audited_account
+                                in str(
+                                    statement["Condition"]["StringEquals"][
+                                        "aws:ResourceAccount"
+                                    ]
+                                )
+                            )
+                            or (
+                                "StringLike" in statement["Condition"]
+                                and "aws:ResourceAccount"
+                                in statement["Condition"]["StringLike"]
+                                and iam_client.audited_account
+                                in str(
+                                    statement["Condition"]["StringLike"][
+                                        "aws:ResourceAccount"
+                                    ]
+                                )
```

Comment:
Please, try to use the `condition_parser` in this check and include these condition keys in the library if not present.

Also update the tests for both the library and the check to cover these new cases.